src/cmakeTaskProvider.ts

```diff
@@ -65,45 +68,66 @@ export class CMakeTask extends vscode.Task {
-        this.defaultTargets = (defaultTargets && defaultTargets.length > 0) ? defaultTargets :
-            this.cmakeDriver ? [this.cmakeDriver.allTargetName] : [allTargetName];
-    }
-
     public async provideTasks(): Promise<CMakeTask[]> {
         const result: CMakeTask[] = [];
-        result.push(await this.provideTask(CommandType.build));
-        result.push(await this.provideTask(CommandType.config));
-        result.push(await this.provideTask(CommandType.install));
-        result.push(await this.provideTask(CommandType.test));
-        result.push(await this.provideTask(CommandType.clean));
-        result.push(await this.provideTask(CommandType.cleanRebuild));
+        const cmakeTools: CMakeTools | undefined = getCMakeToolsForActiveFolder();
+        const cmakeDriver:  CMakeDriver | undefined = (await cmakeTools?.getCMakeDriverInstance()) || undefined;
+        result.push(await this.provideTask(CommandType.build, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.config, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.install, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.test, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.clean, cmakeDriver, cmakeTools?.useCMakePresets));
+        result.push(await this.provideTask(CommandType.cleanRebuild, cmakeDriver, cmakeTools?.useCMakePresets));
         return result;
     }
 
-    public async provideTask(commandType: CommandType): Promise<CMakeTask> {
+    public async provideTask(commandType: CommandType, cmakeDriver?: CMakeDriver, useCMakePresets?: boolean): Promise<CMakeTask> {
         const taskName: string = localizeCommandType(commandType);
+        let targets: string[] | undefined;
+        let preset: string | undefined;
+        const options: { cwd?: string ; environment?: Environment } = {};
+        if (!useCMakePresets) {
+            if (commandType === CommandType.build) {
+                if (cmakeDriver?.targetsName) {
+                    targets = cmakeDriver?.targetsName;
+                } else if (cmakeDriver?.allTargetName) {
+                    targets = [cmakeDriver?.allTargetName];
+                }
+            }
+        } else {
+            switch (commandType) {
+                case CommandType.config:
+                    preset = "${command:cmake.activeConfigurePresetName}";
+                    break;
+                case CommandType.build:
+                    preset = "${command:cmake.activeBuildPresetName}";
+                    break;
+                case CommandType.test:
+                    preset = "${command:cmake.activeTestPresetName}";
+                    break;
+                default:
+                    preset = undefined;
+            }
+            options.cwd = "${workspaceFolder}";
```

Comment:
I believe that all of the drivers default to using the buildDirectory as the cwd for configure and test.  Oddly, it doesn't look like we set it for build.

I believe you can use `${command:cmake.buildDirectory}` to get it.

Reply:
I added the cwd for both preset and non-preset case for all tasks.
it's `${command:cmake.buildDirectory}` now.